### ISSUE TITLE
Is there any fork of TermKit being actively maintained?

### ISSUE THREAD

**Issue by piotr-dobrogost** [unconed/TermKit#74]
I'm asking because it looks like the author of the original project abandoned it.

**Comment by alexcoplan** [comment]
+1 to this would be nice to see an actively maintained fork.

**Comment by cafarm** [comment]
I suggest you look at http://ridiculousfish.com/shell/index.html . It's awesome, actively developed, and great alternative to TermKit.

**Comment by lukestanley** [comment]
Fish didn't work for me ("fish: /lib/i386-linux-gnu/libc.so.6: version `GLIBC_2.15' not found (required by fish)") so I can't compare. The screenshots on the site aren't stunning.

**Comment by alexcoplan** [comment]
@lukestanley would agree there - doesn't really have the same idea as TermKit.

**Comment by cafarm** [comment]
True, fish is less ambitious than TermKit and it's making strides more humbly toward a user-friendly shell. But the reality is fish is useable day-to-day and its got a lot of great features that work now. The same can't be said for TermKit; the idea is great, but it's got a long way to go. We can take the challenge in smaller steps.

**Comment by alexcoplan** [comment]
Actually, just tried fish and it rocks.. although it might not be a termkit, definitely a huge step up from the standard shell and very much useable, thanks.

**Comment by rawsp33d** [comment]
Take a look at [the fork](https://github.com/colinmarc/TermKit) by @colinmarc.  It's not exactly actively maintained, but it actually _worked_ for me (on Ubuntu).  That is far more than I could ever say for this one, considering node.js, socket.io, and every other npm module used have undergone many updates since this repo was last updated.

**Comment by lukestanley** [comment]
That one didn't work for me @rosshadden - node nodekit.js fails just like in this comment: http://blog.easytech.com.ar/2011/05/21/playing-with-termkit-with-chrome/#comment-4582
"node.js:218
throw e; // process.nextTick error, or ‘error’ event on first tick" ... "Error: require.paths is removed. Use node_modules folders, or the NODE_PATH environment variable instead."
Currently unwilling to downgrade node at this time because other things are using it.

**Comment by cooljeanius** [comment]
~~Working on a solution?~~

~~A $5 bounty has been placed on it at [BountySource](https://www.bountysource.com), and you can earn it by having your pull request merged. [https://www.bountysource.com/#repos/unconed/TermKit/issues/74](https://www.bountysource.com/#repos/unconed/TermKit/issues/74)~~
Edit: nvm

**Comment by sohocoke** [comment]
Did @unconed make any announcement regarding abandoning TermKit?

It's such a great idea and with lots of implementation mileage. I'm puzzled how things got so quiet after so much output from him (essays, videos etc).

**Comment by patricknixon** [comment]
+$20 added to [Bounty](https://www.bountysource.com/#repos/unconed/TermKit/issues/74)

**Comment by FireyFly** [comment]
I'm probably not going to actively maintain it given its current state, but I decided to give a go at getting it working (mainly to play around with it a bit myself). This version works for me now, anyway (on Linux, x86_64, via browser): https://github.com/FireyFly/TermKit

**Comment by owais** [comment]
Someone should take this to Kickstarter

**Comment by sohocoke** [comment]
I found this revealing piece of communication, dated 4 months ago:

http://www.reddit.com/r/programming/comments/137kd9/18_months_ago_termkit_a_nextgeneration_terminal/

I'm really interested in projects such as TermKit that challenge entrenched designs in software. It's a shame that @unconed found the reaction to his work so overwhelmingly negative that he has chosen to keep 'twiddling' with it out of sight. 

Maybe we'll eventually see great things here.

**Comment by tehprofessor** [comment]
I for one would love to see this get some active development. I'd even be happy to lend a hand even though I hate node (well within reason, it does some things well; JS in general just makes my eyes bleed from time to time).

It took me ages of searching to even figure out the name of this project (I fogot)!

Perhaps it's time to get some of that hard core forking action on...

**Comment by axaluss** [comment]
Hi what about https://github.com/p-e-w/finalterm

**Comment by nmcv** [comment]
@axaluss Seen anybody use it on OSX? The requirements list is pretty bad for OSX brewers ): Nice suggestion though, installed it in Linux.

**Comment by krilnon** [comment]
What exactly is this bounty for?  The issue is about maintaining a fork of this project, but it kind of seems like the bounty is more about getting the existing code into some sort of runnable shape.

**Comment by Jonahss** [comment]
I just now thought to myself:
 "I'd like to build a terminal emulator on node webkit so I could have more visual widgets for displaying things".
"I bet that already exists"
Then I found termkit, and it's exactly what i envisioned.
Then I found out it's dead.

What do?

**Comment by axaluss** [comment]
Term Kit is dead because the founder encountered too much criticism. 
Feel free to improve the project.

**Comment by Jonahss** [comment]
@axaluss that's dispiriting. What was all the criticism about? The fact that there's so much attention still on this repo shows that it appealed to some people.

Was it the same sort of response that vim/emacs users give to anybody who uses a visual editor?

**Comment by booc0mtaco** [comment]
I'd like to know as well. I remember being excited about this project, then it got quiet a few years ago...

Somewhat related... There's this app. It has some of the same interesting features and interactivity I remember : https://www.kickstarter.com/projects/xiki/xiki-the-command-revolution

**Comment by rekado** [comment]
There is eshell which comes with Emacs.  It can display inline images (also in combination with `iimage-mode`) and can be greatly extended in elisp.  Any elisp function defined in Emacs is available right there in eshell.

@Jonahss 

> Was it the same sort of response that vim/emacs users give to anybody who uses a visual editor?

Emacs is a visual editor unless you run it in a terminal, which is weird, because with Emacs you rather run the terminal inside of Emacs.  Strings in Emacs can have `text-properties` that can radically alter the way text is dislayed (e.g. by replacing it with an image).

**Comment by pirate** [comment]
I'm sad that it's 2020 and there's still no good alternative that has carried on with this idea.  I still think about TermKit regularly and it's been almost a decade since I first saw it.

**Comment by Jonahss** [comment]
There's [Hyper](https://hyper.is/), it works, but no new developments have been made in a few years.

**Comment by martonlanga** [comment]
I'm working on [Termy](https://github.com/termyapp/termy). For now, it's still missing a lot of features, but it will have autosuggestions, a frontend for displaying rich content (HTML), structured data piping and many others.

You can try it out on [Mac](https://github.com/termyapp/termy/releases/tag/v0.1.0). I would love to know whether the app runs properly and running commands work.

There is also a [Discord channel](http://discord.com/invite/tzrRhdZ) if you'd like to chat!

**Comment by cooljeanius** [comment]
> I'm working on [Termy](https://github.com/termyapp/termy).

This link 404s; is there a replacement?

**Comment by pirate** [comment]
Here is a list of the "modern re-imagined terminal emulator" projects I know about as of 2024:

- https://www.warp.dev/
- https://hyper.is/
- https://terminal.click/
- https://tabby.sh/
- https://www.waveterm.dev/
- https://termius.com/
- https://raphamorim.io/rio/
- https://sw.kovidgoyal.net/kitty/
- https://kingtoolbox.github.io/ aka https://github.com/kingToolbox/WindTerm
- https://github.com/Swordfish90/cool-retro-term
- https://github.com/microsoft/terminal (Windows only)
- [Ghostty](https://x.com/mitchellh/status/1662217955424493570) (private Beta)

**Older Options:**  
- https://wezfurlong.org/wezterm/
- https://alacritty.org/
- https://www.emtec.com/zoc/features.html
- https://iterm2.com/downloads.html
- https://www.macterm.net/
- https://upterm.dev/
- https://github.com/gnome-terminator/terminator

More options here: https://github.com/cdleon/awesome-terminals
